src/Runner.Worker/Handlers/CompositeActionHandler.cs

```diff
@@ -44,7 +68,28 @@ public async Task RunAsync(ActionRunStage stage)
                 {
                     inputsData[i.Key] = new StringContextData(i.Value);
                 }
+                
+                List<JobExtensionRunner> containerDownloadSteps = null;
+                
+                // Download actions and expand graph for local actions
+                if (Action is Pipelines.RepositoryPathReference repoAction &&
+                    string.Equals(repoAction.RepositoryType, Pipelines.PipelineConstants.SelfAlias, StringComparison.OrdinalIgnoreCase))
+                {
 
+                    var actionManager = HostContext.GetService<IActionManager>();
+                    var prepareResult = await actionManager.PrepareActionsAsync(ExecutionContext, steps, ExecutionContext.Id);
+                    foreach (var step in prepareResult.ContainerSetupSteps)
+                    {
+                        if (step is JobExtensionRunner)
```

Comment:
i wish one day we can name this something like "SystemStepRunner" (unrelated to your PR)

Reply:
because we dont have different job types anymore. so this naming is based on legacy stuff that doesnt exist anymore

Reply:
Going to address this in a later pr